typescript/listeners/idle.py

```diff
@@ -179,8 +181,46 @@ def update_status(self, view, info):
         else:
             view.run_command('typescript_quick_info')
 
+    def request_document_highlights(self, view, info):
+        if is_typescript(view):
+            location = get_location_from_view(view)
+            cli.service.async_document_highlights(view.file_name(), location, self.highlight_occurrences)
+
+    def highlight_occurrences(self, response):
+        view = active_view()
+        if not view.file_name():
+            return
+
+        region_key = 'occurrences'
+        view.erase_regions(region_key)
+
+        if not response['success']:
+            return
+
+        occurrence_regions = []
+
+        for file_highlight in response['body']:
+            if file_highlight['file'] != view.file_name().replace('\\', '/'):
```

Comment:
If I remember correctly, the tsserver changes the file name t olower case if the OS' file system is case-insensitive. Would that cause problems here? for example on or linux. Also, as we are comparing filenames in multiple places, it might worth it to extract that part to a individual function.